advanced_source/super_resolution_with_onnxruntime.py

```diff
@@ -0,0 +1,285 @@
+                               input_names = ['input'],   # the model's input names
+                               output_names = ['output'], # the model's output names
+                               dynamic_axes={'input' : {0 : 'batch_size'},    # variable lenght axes
+                                             'output' : {0 : 'batch_size'}})
+
+######################################################################
+# ``torch_out`` is the output after executing the model. Normally you can
+# ignore this output, but here we will use it to verify that the model we
+# exported computes the same values when run in onnxruntime.
+#
+# But before verifying the model's output with onnxruntime, we will check
+# the onnx model with onnx's API. This will verify the model's structure
+# and confirm that the model has a valid schema. 
+
+import onnx
+
+onnx_model = onnx.load("super_resolution.onnx")
+onnx.checker.check_model(onnx_model)
+
+
+######################################################################
+# Now let's create an onnxruntime session. This part can normally be
+# done in a separate process or on another machine, but we will
+# continue in the same process so that we can verify that onnxruntime
+# and PyTorch are computing the same value for the network:
+#
+
+import onnxruntime
+
+ort_session = onnxruntime.InferenceSession("super_resolution.onnx")
+
+def to_numpy(tensor):
+    return tensor.detach().cpu().numpy() if tensor.requires_grad else tensor.cpu().numpy()
+
+# compute onnxruntime output prediction
+ort_inputs = {ort_session.get_inputs()[0].name: to_numpy(x)}
+ort_outs = ort_session.run(None, ort_inputs)
+
+# compare onnxruntime and PyTorch results
+np.testing.assert_allclose(to_numpy(torch_out), ort_outs[0], rtol=1e-03, atol=1e-05)
+
+print("Exported model has been tested with ONNXRuntime, and the result looks good!")
+
+
+######################################################################
+# We should see that the output of PyTorch and onnxruntime runs match
+# numerically with the given precision (rtol=1e-03 and atol=1e-05).
+# As a side-note, if they do not match then there is an issue in the
+# onnx exporter, so please contact us in that case.
+#
+
+
+######################################################################
+# Transfering SRResNet using ONNX
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+#
+
+
+######################################################################
+# Using the same process as above, we also transferred an interesting new
+# model "SRResNet" for super-resolution presented in `this
+# paper <https://arxiv.org/pdf/1609.04802.pdf>`__ (thanks to the authors
+# at Twitter for providing us code and pretrained parameters for the
+# purpose of this tutorial). The model definition and a pre-trained model
+# can be found
+# `here <https://gist.github.com/prigoyal/b245776903efbac00ee89699e001c9bd>`__.
+# Below is what SRResNet model input, output looks like. |SRResNet|
```

Comment:
This model was only mentioned in the previous version. 

We are able to export it as well and match numbers with ORT.
Should I add the exported .onnx model or just mention it like before?

Reply:
I don't think model will help us show more details, let's keep it.

Reply:
you mean remove the whole section?

Reply:
Yeah... two models are quite similar.. I didn't see any benefits here... Just remove it... of course, follow the same process as above, we will be able to transfer tons of models :-)